inc/fields/class-field-user-select.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+class Shortcode_UI_Field_User_Select {
+
+	private static $instance;
+
+	// All registered user fields.
+	private $user_fields  = array();
+
+	// Field Settings.
+	private $fields = array(
+		'user_select' => array(
+			'template' => 'shortcode-ui-field-user-select',
+			'view'     => 'editAttributeFieldUserSelect',
+		),
+	);
+
+	/**
+	 * Setup the instance.
+	 *
+	 * @return Shortcode_UI_Field_User_Select
+	 */
+	public static function get_instance() {
+
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new self;
+			self::$instance->setup_actions();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Add the required actions and filters.
+	 */
+	private function setup_actions() {
+
+		add_filter( 'shortcode_ui_fields',             array( $this, 'filter_shortcode_ui_fields' ) );
+		add_action( 'enqueue_shortcode_ui',            array( $this, 'action_enqueue_shortcode_ui' ) );
+		add_action( 'wp_ajax_shortcode_ui_user_field', array( $this, 'action_wp_ajax_shortcode_ui_user_field' ) );
+		add_action( 'shortcode_ui_loaded_editor',      array( $this, 'action_shortcode_ui_loaded_editor' ) );
+	}
+
+	/**
+	 * Add our field to the shortcode fields.
+	 *
+	 * @param $fields
+	 *
+	 * @return array
+	 */
+	public function filter_shortcode_ui_fields( $fields ) {
+
+		return array_merge( $fields, $this->fields );
+	}
+
+	/**
+	 * Add Select2 for our UI.
+	 */
+	public function action_enqueue_shortcode_ui() {
+
+		$plugin_dir = dirname( dirname( __FILE__ ) );
+
+		wp_enqueue_script( 'select2', plugins_url( 'lib/select2/select2.min.js', $plugin_dir ), array( 'jquery', 'jquery-ui-sortable' ), '3.5.2' );
+		wp_enqueue_style( 'select2', plugins_url( 'lib/select2/select2.css', $plugin_dir ), null, '3.5.2' );
```

Comment:
Should `select2` be registered (e.g. via `wp_register_script()`) once and then just the handle enqueued here?

Here's how I've been doing it for a Customizer control that uses Select2:
https://github.com/xwp/wp-customize-object-selector/blob/423e8f62a326ba05f164b0077ce944ad2ed91bfc/php/class-plugin.php#L70-L76
https://github.com/xwp/wp-customize-object-selector/blob/423e8f62a326ba05f164b0077ce944ad2ed91bfc/php/class-plugin.php#L106-L112